autogpt_platform/backend/backend/data/graph.py

```diff
@@ -328,7 +329,7 @@ def from_db(graph: AgentGraph, hide_credentials: bool = False):
 
     @staticmethod
     def _process_node(node: AgentNode, hide_credentials: bool) -> NodeModel:
-        node_dict = node.model_dump()
+        node_dict = {field: getattr(node, field) for field in node.model_fields}
```

Comment:
? @majdyz

Reply:
This piece of code dumps a Prisma model and then parses it again, aside from being excessively expensive, unfortunately not all the objects is serializable, including the new information added in AgentNode (or it's child). So it breaks stuff.

Reply:
What non-serializable fields are you referring to?

Reply:
One of the thing I remember is a Json column of this prisma object. When it's dumped and reparsed it became a normal dict, not a prisma json object and you can't instantiate the prisma object back.

Reply:
Ah yeah Prisma requires explicitly casting or converting between `dict` and `Json`. Not ideal.

Reply:
Can you just do prisma’s Json object? I think I had to for that email script

Reply:
There is also reference table fields for foreign keys. Either way, dump & parse is not the way to go. Especially for this simple field excluding operation.